### PR TITLE
fix canEdit for category, add customized tag feature

### PR DESCRIPTION
**Pull request by louisewang1**
image

- fix the canEdit attribute for category selection, when user don't have editting right, the menu will be disabled
- add a input form to set customized tags for the item (in pic above)
- age range => level
- Now options for disciplines are ordered alphabetically, while options for levels maintained the logical order

### CODE REVIEW

**Comment by louisewang1** [comment]
close #252 
close #245 
close #247

File: src/components/item/sharing/CustomizedTags.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+          onChange={handleChange}
+        />
+        <Button
+          type="submit"
+          variant="outlined"
+          color="primary"
+          className={classes.button}
+          endIcon={<SendIcon />}
+        >
+          {t('Submit')}
```
Comment (pyphilia) [inline src/components/item/sharing/CustomizedTags.js:89]: I would actually use `save` instead, what do you think? (and change the icon or remove it accordingly)

File: src/components/item/sharing/CustomizedTags.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        {t('Customized Tags')}
+      </Typography>
+      <Typography variant="body1">
+        {t('Please seperate tags by comma. ')}
+        {t('Eg. Tag1,Tag2,Tag3,...,TagN')}
+      </Typography>
+      <form className={classes.container} onSubmit={handleSubmit}>
+        <TextField
+          disabled={!edit}
+          id="customized-tags-input-box"
```
Comment (pyphilia) [inline src/components/item/sharing/CustomizedTags.js:74]: Unless this refers to an element, remove it.

File: src/components/item/sharing/CustomizedTags.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  };
+
+  return (
+    <>
+      <Typography variant="h6" className={classes.title}>
+        {t('Customized Tags')}
+      </Typography>
+      <Typography variant="body1">
+        {t('Please seperate tags by comma. ')}
+        {t('Eg. Tag1,Tag2,Tag3,...,TagN')}
```
Comment (pyphilia) [inline src/components/item/sharing/CustomizedTags.js:69]: Very engineer like ;) Maybe let's go for real example tags: `biology`, `moon`, ...

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks for this PR! 
I see your branch is still called 237/categories, I think it should be another branch refering to another issue? It's okay for this time, but keep in mind to create new branches for each new issue.
What do you think of `Tags` only instead of `Customized Tags`?

**Comment by louisewang1** [comment]
If I use "Tags", would this be confused with itemTags (like "published", "private")? Besides that I'm good with just "Tags".

File: src/components/item/sharing/CategorySelection.js [hunk 4]
```diff
@@ -62,12 +64,16 @@ function CategorySelection({ item, edit }) {
...
     ?.toArray();
   const disciplineList = categoriesMap
-    ?.get(categoryTypes?.filter((type) => type.name === 'discipline').get(0).id)
-    ?.toArray();
+    ?.get(
+      categoryTypes?.find((type) => type.name === CATEGORY_TYPES.DISCIPLINE)
+        ?.id,
+    )
+    ?.toArray()
+    .sort(compare);
```
Comment (pyphilia) [inline src/components/item/sharing/CategorySelection.js:76]: I would rename the sorting function to something more explicit.

File: src/components/item/sharing/CategorySelection.js [hunk 5]
```diff
@@ -123,55 +129,51 @@ function CategorySelection({ item, edit }) {
...
+          value={levelList?.filter((value) => selectedValues.includes(value))}
+          getOptionSelected={(option, value) => option.id === value.id}
+          options={levelList}
+          getOptionLabel={(option) => option.name}
+          onChange={handleChange('level')}
+          renderInput={(params) => (
+            <TextField
+              {...params}
+              variant="outlined"
+              placeholder={t('Please choose from list')}
```
Comment (pyphilia) [inline src/components/item/sharing/CategorySelection.js:151]: `Please choose from the list`?

File: src/components/item/sharing/CustomizedTagsEdit.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  };
+
+  return (
+    <>
+      <Typography variant="h6" className={classes.title}>
+        {t('Tags')}
+      </Typography>
+      <Typography variant="body1">
+        {t('Please seperate tags by comma. ')}
+        {t('Eg. English,Biology,Lab,Plants,...,Demo')}
```
Comment (pyphilia) [inline src/components/item/sharing/CustomizedTagsEdit.js:69]: Add spaces between values

**Review by pyphilia — CHANGES_REQUESTED** [review]
Nice PR! I left some comments 🐰

**Pull request merged**